core/services/vrf/delegate.go

```diff
@@ -357,13 +537,14 @@ func GetVRFInputs(jb job.Job, request *solidity_vrf_coordinator_interface.VRFCoo
 func (lsn *listener) Close() error {
 	return lsn.StopOnce("VRFListener", func() error {
 		close(lsn.chStop)
-		<-lsn.waitOnStop
+		<-lsn.waitOnStop // Log listener
+		<-lsn.waitOnStop // Head listener
```

Comment:
could use a waitgroup? seems to be a pattern in other services

Reply:
yeah I was going to but its actually nice to have the channel for testability https://github.com/smartcontractkit/chainlink/blob/4803aba4099ab2f7b2e800891fc1d343a4ed0285/core/services/vrf/delegate_test.go#L163